<launch>

  <arg name="align_depth"               default="false"/>
  <arg name="unite_imu_method"          default="linear_interpolation"/>
  <arg name="enable_gyro"               default="true"/>
  <arg name="enable_accel"              default="true"/>

  <arg name="enable_color"              default="false"/>
  <arg name="publish_odom_tf"           default="true"/>  
  <arg name="enable_depth"              default="false"/>
  <arg name="filters"                   default="pointcloud"/>



  <!-- <arg name="map_file" default="/home/ssapl/autobot/src/taurus_data/map/sick_map.yaml"/>

  <arg name="urdf_file" default="$(find realsense2_description)/urdf/test_d435i_camera.urdf.xacro" />
  <param name="robot_description" command="$(find xacro)/xacro - -inorder $(arg urdf_file)" />

  <node name="robot_state_publisher" pkg="robot_state_publisher"
        type="robot_state_publisher">
    <param name="publish_frequency" value="30.0"/>
  </node> -->

  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="enable_color"             value="$(arg enable_color)"/>
    <arg name="publish_odom_tf"          value="$(arg publish_odom_tf)"/>  
    <arg name="enable_depth"             value="$(arg enable_depth)"/>
    <!-- <arg name="filters"                  value="$(arg filters)"/> -->
    <arg name="align_depth"              value="$(arg align_depth)"/>
    <arg name="unite_imu_method"         value="$(arg unite_imu_method)"/>
    <arg name="enable_gyro"              value="$(arg enable_gyro)"/>
    <arg name="enable_accel"             value="$(arg enable_accel)"/>
    <param name="initial_reset"          value="true"/>
  </include>

  
  <node pkg="imu_filter_madgwick" type="imu_filter_node"
      name="imu_filter">
    <param name="use_mag" value="False" />
    <param name="publish_tf" value="true" />
    <param name="fixed_frame" value="base_link"/>
    <param name="remove_gravity_vector" value="true" /> 

    <remap from="imu/data_raw" to="camera/imu_t" />
    <remap from="imu/data" to="filtered/imu_map" />
  </node>

  <!-- <node pkg="bot" name="imu_filter2" type="imu_filter.py"/> -->

  <!-- <include file="$(find rtabmap_ros)/launch/rtabmap.launch">
    <arg name="rtabmap_args"  value="- -delete_db_on_start - -Optimizer/GravitySigma 0.3"/>
    <arg name="depth_topic"    value="/camera/aligned_depth_to_color/image_raw"/>
    <arg name="rgb_topic"    value="/camera/color/image_raw"/>
    <arg name="camera_info_topic"    value="/camera/color/camera_info"/>
    <arg name="approx_sync"    value="false"/>
    <arg name="wait_imu_to_init"    value="true"/>
    <arg name="imu_topic"    value="/rtabmap/imu"/>
  </include> -->

  <!-- <node name="ekf_localization" pkg="robot_localization" type="ekf_localization_node" args="$(arg map_file)" >
    <rosparam command="load" file="$(find bot)/config/imu_ekf.yaml" />
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find bot)/config/rs_imu.rviz"/> -->

 
</launch>



